jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -1824,12 +1824,23 @@ ecma_op_function_list_lazy_property_names (ecma_object_t *object_p, /**< functio
 {
 #if ENABLED (JERRY_ESNEXT)
   ecma_extended_object_t *ext_func_p = (ecma_extended_object_t *) object_p;
-  if (!ECMA_GET_FIRST_BIT_FROM_POINTER_TAG (ext_func_p->u.function.scope_cp))
+  ecma_value_t prop_name_val = ecma_make_magic_string_value (LIT_MAGIC_STRING_LENGTH);
+  if (!ECMA_GET_FIRST_BIT_FROM_POINTER_TAG (ext_func_p->u.function.scope_cp)
+      || ecma_op_ordinary_object_has_own_property (object_p, ecma_op_to_string (prop_name_val)))
```

Comment:
This means, if the property is created, and deleted and created again it will appear early. Are you sure this is correct?

Reply:
Good point! You are right, if we delete the `length` property and defined later. Then `length`  should be the end of the list.